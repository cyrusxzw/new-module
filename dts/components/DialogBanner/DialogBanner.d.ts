export function DialogBanner({
  className,
  copy,
  cta,
  hasCloseButton,
  heading,
  isVisible,
  message,
  onClose,
  position,
  theme,
}: {
  className: any;
  copy: any;
  cta: any;
  hasCloseButton: any;
  heading: any;
  isVisible: any;
  message: any;
  onClose: any;
  position: any;
  theme: any;
}): JSX.Element;
export namespace DialogBanner {
  namespace propTypes {
    const className: PropTypes.Requireable<string>;
    const copy: PropTypes.Requireable<
      PropTypes.InferProps<{
        closeButtonTitle: PropTypes.Requireable<string>;
      }>
    >;
    const cta: PropTypes.Requireable<object>;
    const hasCloseButton: PropTypes.Requireable<boolean>;
    const isVisible: PropTypes.Requireable<boolean>;
    const message: PropTypes.Validator<string>;
    const onClose: PropTypes.Validator<(...args: any[]) => any>;
    const position: PropTypes.Requireable<string>;
    const theme: PropTypes.Requireable<string>;
    const heading: PropTypes.Validator<string>;
  }
  namespace defaultProps {
    const className_1: any;
    export { className_1 as className };
    export namespace copy_1 {
      const closeButtonTitle: string;
    }
    export { copy_1 as copy };
    const cta_1: any;
    export { cta_1 as cta };
    const hasCloseButton_1: boolean;
    export { hasCloseButton_1 as hasCloseButton };
    const isVisible_1: boolean;
    export { isVisible_1 as isVisible };
    const message_1: any;
    export { message_1 as message };
    const onClose_1: any;
    export { onClose_1 as onClose };
    const position_1: string;
    export { position_1 as position };
    const theme_1: string;
    export { theme_1 as theme };
    const heading_1: any;
    export { heading_1 as heading };
  }
}
import PropTypes from 'prop-types';
