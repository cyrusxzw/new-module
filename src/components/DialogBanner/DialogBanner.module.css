@import 'styles/variables/index';
@import 'styles/mixins/index';

.base {
  position: fixed;
  z-index: var(--z-index-fixed-dialog-banner);
  right: 0;
  left: 0;
  padding: 20px 40px;

  @media (--viewport-xs-sm-only) {
    padding: 20px 20px 40px 20px;
  }

  @media (--viewport-md) {
    display: grid;
    grid-template-columns: 15% 63% 22%;
  }

  @media (--viewport-lg) {
    grid-template-columns: 20% 60% 20%;
  }
}

.top {
  top: 0;
}

.bottom {
  @extend %slideOutDown;

  bottom: 0;
}

/* @TODO Handle bottom prop */
.isVisible {
  &.bottom {
    @extend %slideInUp;
  }
}

.dark {
  background-color: var(--color-secondary);
}

.black {
  background-color: var(--color-grey-10);
}

.contentContainer {
  @media (--viewport-md) {
    padding: 0 24px;
  }
}

.heading {
  color: var(--color-white);

  @media (--viewport-xs-sm-only) {
    margin-top: 28px;
    margin-bottom: 10px;
  }
}

.message {
  color: var(--color-grey-65);
  font-size: rem(15px);

  @media (--viewport-xs-sm-only) {
    &:last-of-type {
      margin-bottom: 14px;
    }
  }
}

.warningIcon {
  fill: var(--color-grey-50);
}

.closeButton {
  position: absolute;
  top: 20px;
  right: 40px;

  @media (--viewport-xs-sm-only) {
    top: 20px;
    right: 20px;
  }

  &:hover,
  &:focus {
    svg {
      fill: var(--color-white);
    }
  }
}

.closeIcon {
  fill: var(--color-grey-50);
}

.closeContainer {
  @media (--viewport-md) {
    align-self: flex-end;
    text-align: right;
  }
}

.ctaButton {
  font-size: rem(15px);
}
