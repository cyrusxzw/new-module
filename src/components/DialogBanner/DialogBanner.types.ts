import type { VFC } from 'react';

type DialogBannerProps = {
  className?: string;
  copy?: {
    closeButtonTitle?: string;
  };
  cta?: {
    dataTestRef?: string;
    id?: string;
    openInANewWindow?: boolean;
    text?: string;
    title?: string;
    url: string;
  };
  hasCloseButton?: boolean;
  isVisible?: boolean;
  message: string;
  onClose: () => void;
  position?: 'top' | 'bottom';
  theme?: 'dark' | 'black';
  heading: string;
};

export type DialogBannerType = VFC<DialogBannerProps>;
