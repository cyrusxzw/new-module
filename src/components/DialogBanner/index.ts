import { DialogBanner } from './DialogBanner';

export { DialogBanner };
