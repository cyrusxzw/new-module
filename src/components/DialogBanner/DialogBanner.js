import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import { useEscapeKeyListener, useOverflowHidden } from '~/customHooks';
import { Button } from '~/components/Button';
import { Heading } from '~/components/Heading';
import { Hyperlink } from '~/components/Hyperlink';
import { Icon } from '~/components/Icon';
import { P } from '~/components/Paragraph';
import { Overlay } from '~/components/Overlay';
import { Transition } from '~/components/Transition';
import styles from './DialogBanner.module.css';

const DialogBanner = ({
  className,
  copy,
  cta,
  hasCloseButton,
  heading,
  isVisible,
  message,
  onClose,
  position,
  theme,
}) => {
  useEscapeKeyListener(onClose);
  useOverflowHidden(isVisible);

  const classSet = cx(
    styles.base,
    { [styles.isVisible]: isVisible },
    styles[theme],
    styles[position],
    className,
  );

  return (
    <>
      <Overlay isVisible={isVisible} onClose={onClose} />
      <Transition
        isActive={isVisible}
        shouldMountOnEnter={true}
        shouldUnmountOnExit={true}
        type="slideDown"
      >
        <aside className={classSet}>
          <div className={styles.iconContainer}>
            <Icon
              className={styles.warningIcon}
              height={22}
              name="warning"
              theme="light"
              width={22}
            />
          </div>
          <div className={styles.contentContainer}>
            <Heading
              className={styles.heading}
              level="1"
              noTopMargin={true}
              size="large"
            >
              {heading}
            </Heading>
            <P className={styles.message}>{message}</P>
          </div>
          <div className={styles.closeContainer}>
            {hasCloseButton && (
              <Button
                className={styles.closeButton}
                isInline={true}
                onClick={onClose}
                title={copy.closeButtonTitle}
              >
                <Icon
                  className={styles.closeIcon}
                  height={12}
                  name="close"
                  theme="light"
                  width={12}
                />
              </Button>
            )}
            {cta && cta.text && (
              <Hyperlink
                className={styles.ctaButton}
                dataTestRef={cta.dataTestRef}
                id={cta.id}
                isInline={true}
                target={cta.openInANewWindow ? '_blank' : '_self'}
                theme={'light'}
                title={cta.title}
                url={cta.url}
              >
                {cta.text}
              </Hyperlink>
            )}
          </div>
        </aside>
      </Transition>
    </>
  );
};

DialogBanner.propTypes = {
  className: PropTypes.string,
  copy: PropTypes.shape({
    closeButtonTitle: PropTypes.string,
  }),
  cta: PropTypes.object,
  hasCloseButton: PropTypes.bool,
  isVisible: PropTypes.bool,
  message: PropTypes.string.isRequired,
  onClose: PropTypes.func.isRequired,
  position: PropTypes.oneOf(['top', 'bottom']),
  theme: PropTypes.oneOf(['dark', 'black']),
  heading: PropTypes.string.isRequired,
};

DialogBanner.defaultProps = {
  className: undefined,
  copy: {
    closeButtonTitle: 'Close Dialog Banner',
  },
  cta: undefined,
  hasCloseButton: true,
  isVisible: false,
  message: undefined,
  onClose: undefined,
  position: 'top',
  theme: 'dark',
  heading: undefined,
};

export { DialogBanner };
