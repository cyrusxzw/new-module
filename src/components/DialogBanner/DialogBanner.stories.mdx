import { Fragment } from 'react';
import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks';
import { State } from '~.storybook/State';
import * as knobs from '@storybook/addon-knobs';
import { Button } from '~/components/Button';
import { DialogBanner } from './DialogBanner';

<Meta title="Components/DialogBanner" component={DialogBanner} />

# DialogBanner

<Canvas>
  <Story name="Base component">
    <State initialState={{ isVisible: false }}>
      {(state, setState) => (
        <Fragment>
          <DialogBanner
            cta={{
              text: 'Vestibulum ullamcorper.',
              url: '/',
            }}
            heading={knobs.text(
              'heading',
              'Maecenas elementum, metus id facilisis',
              'Copy',
            )}
            isVisible={state.isVisible}
            message={knobs.text(
              'message',
              'Nullam pretium, risus a dictum rhoncus, ex libero cursus erat, dignissim malesuada tortor justo quis enim. Nulla pellentesque sit amet ante et molestie.',
              'Copy',
            )}
            onClose={() => setState({ ...state, isVisible: false })}
            position={knobs.select(
              'position',
              ['top', 'bottom'],
              'top',
              'Presentation',
            )}
            hasCloseButton={knobs.boolean(
              'hasCloseButton',
              true,
              'Presentation',
            )}
            theme={knobs.select(
              'theme',
              ['dark', 'black'],
              'dark',
              'Presentation',
            )}
          />
          <Button
            onClick={() => setState({ ...state, isVisible: true })}
            title="Open Dialog Banner"
          >
            Open Dialog Banner
          </Button>
        </Fragment>
      )}
    </State>
  </Story>
</Canvas>

## API

<ArgsTable of={DialogBanner} />
