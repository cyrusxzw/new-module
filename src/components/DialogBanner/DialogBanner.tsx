import React from 'react';
import cx from 'classnames';
import { useEscapeKeyListener, useOverflowHidden } from '~/customHooks';
import { Button } from '~/components/Button';
import { Heading } from '~/components/Heading';
import { Hyperlink } from '~/components/Hyperlink';
import { Icon } from '~/components/Icon';
import { P } from '~/components/Paragraph';
import { Overlay } from '~/components/Overlay';
import { Transition } from '~/components/Transition';
import type { DialogBannerType } from './DialogBanner.types';
import styles from './DialogBanner.module.css';

const DialogBanner: DialogBannerType = ({
  className,
  copy,
  cta,
  hasCloseButton = true,
  isVisible = false,
  message,
  onClose,
  position = 'top',
  theme = 'dark',
  heading,
}) => {
  useEscapeKeyListener(onClose);
  useOverflowHidden(isVisible);

  const classSet = cx(
    styles.base,
    { [styles.isVisible]: isVisible },
    styles[theme],
    styles[position],
    className,
  );

  return (
    <>
      <Overlay isVisible={isVisible} onClose={onClose} />
      <Transition
        isActive={isVisible}
        shouldMountOnEnter={true}
        shouldUnmountOnExit={true}
        type="slideDown"
      >
        <aside className={classSet}>
          <div className={styles.iconContainer}>
            <Icon
              className={styles.warningIcon}
              height={22}
              name="warning"
              theme="light"
              width={22}
            />
          </div>
          <div className={styles.contentContainer}>
            <Heading className={styles.heading} level="1" size="large">
              {heading}
            </Heading>
            <P className={styles.message}>{message}</P>
          </div>
          <div className={styles.closeContainer}>
            {hasCloseButton && (
              <Button
                className={styles.closeButton}
                isInline={true}
                onClick={onClose}
                title={copy?.closeButtonTitle ?? 'Close Dialog Banner'}
              >
                <Icon
                  className={styles.closeIcon}
                  height={12}
                  name="close"
                  theme="light"
                  width={12}
                />
              </Button>
            )}
            {cta && cta.text && (
              <Hyperlink
                className={styles.ctaButton}
                dataTestRef={cta.dataTestRef}
                hasTargetInNewWindow={cta.openInANewWindow}
                id={cta.id}
                theme={'light'}
                title={cta.title}
                url={cta.url}
              >
                {cta.text}
              </Hyperlink>
            )}
          </div>
        </aside>
      </Transition>
    </>
  );
};

export { DialogBanner };
