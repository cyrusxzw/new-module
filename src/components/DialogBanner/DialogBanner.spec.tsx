import React from 'react';
import { render } from '@testing-library/react';
import { DialogBanner } from './DialogBanner';

const mockFn = jest.fn();

describe('<DialogBanner />', () => {
  it('should render base component correctly', () => {
    const { container } = render(
      <DialogBanner
        cta={{
          text: 'Need assistance?',
          url: '/contact-us/',
        }}
        hasCloseButton={true}
        heading="Maximum allowed quantity exceeded."
        isVisible={true}
        message="Our apologies, You have exceeded the maximum allowed quantity of this item."
        onClose={mockFn}
        position="top"
        theme="dark"
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
