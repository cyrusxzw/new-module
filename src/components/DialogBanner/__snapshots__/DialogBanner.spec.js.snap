// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DialogBanner /> should render base component correctly 1`] = `
<div>
  <div
    aria-hidden="true"
    class="base isVisible"
    data-testid="data-testid-Overlay"
    role="button"
    tabindex="-1"
  />
  <aside
    class="base isVisible dark top"
  >
    <div
      class="iconContainer"
    >
      <svg
        aria-labelledby="warning-1"
        class="base warning light warningIcon"
        data-testid="data-testid-Icon"
        focusable="false"
        height="22"
        role="img"
        style="height: 22px; width: 22px;"
        tabindex="-1"
        viewBox="0 0 50 50"
        width="22"
      >
        <title
          id="warning-1"
        >
          warning
        </title>
        <g>
          <path
            d="M42.7,7.2C37.7,2.2,31.5,0,25,0S12.3,2.2,7.3,7.2c-9.7,9.7-9.7,25.4,0,35.2c5,5,11.2,7.2,17.7,7.2s12.7-2.5,17.7-7.2 C52.4,32.6,52.4,16.9,42.7,7.2z M41,40.6c-4.2,4.2-10,6.5-16,6.5s-11.7-2.2-16-6.5c-8.7-8.7-8.7-23,0-31.7c4.3-4.2,10-6.5,16-6.5 s11.7,2.3,16,6.5C49.7,17.6,49.7,31.9,41,40.6z"
          />
          <path
            d="M27,13.1V30h-4V13.1H27z M23,36.5h3.9v-4.3H23V36.5z"
          />
        </g>
      </svg>
    </div>
    <div
      class="contentContainer"
    >
      <h1
        class="base large dark heading"
      >
        Maximum allowed quantity exceeded.
      </h1>
      <p
        class="base dark message"
      >
        Our apologies, You have exceeded the maximum allowed quantity of this item.
      </p>
    </div>
    <div
      class="closeContainer"
    >
      <button
        class="base dark closeButton"
        title="Close Dialog Banner"
        type="button"
      >
        <svg
          aria-labelledby="close-2"
          class="base close light closeIcon"
          data-testid="data-testid-Icon"
          focusable="false"
          height="12"
          role="img"
          style="height: 12px; width: 12px;"
          tabindex="-1"
          viewBox="0 0 50 50"
          width="12"
        >
          <title
            id="close-2"
          >
            close
          </title>
          <g>
            <polygon
              class="close"
              points="50,5 45,0 25,20 5,0 0,5 20,25 0,45 5,50 25,30 45,50 50,45 30,25"
            />
          </g>
        </svg>
      </button>
      <a
        class="base left light ctaButton"
        href="/contact-us/"
        target="_self"
      >
        Need assistance?
      </a>
    </div>
  </aside>
</div>
`;
